include/nbl/builtin/glsl/ext/FFT/fft.glsl

```diff
@@ -81,71 +79,86 @@ vec2 nbl_glsl_ext_FFT_getPaddedData(in uvec3 coordinate, in uint channel);
 #error "You need to define `nbl_glsl_ext_FFT_getPaddedData` and mark `_NBL_GLSL_EXT_FFT_GET_PADDED_DATA_DEFINED_`!"
 #endif
 
-// Count Leading Zeroes (naive?)
-uint nbl_glsl_ext_FFT_clz(in uint x) 
-{
-    return 31u - findMSB(x);
-}
-
-uint nbl_glsl_ext_FFT_reverseBits(in uint x)
-{
-    return bitfieldReverse(x);
-}
-
 uint nbl_glsl_ext_FFT_calculateTwiddlePower(in uint threadId, in uint iteration, in uint logTwoN) 
 {
     const uint shiftSuffix = logTwoN - 1u - iteration; // can we assert that iteration<logTwoN always?? yes
     const uint suffixMask = (2u << iteration) - 1u;
     return (threadId & suffixMask) << shiftSuffix;
 }
 
-nbl_glsl_complex nbl_glsl_ext_FFT_twiddle(in uint threadId, in uint iteration, in uint logTwoN, in uint N) 
+nbl_glsl_complex nbl_glsl_ext_FFT_twiddle(in uint threadId, in uint iteration, in uint logTwoN) 
 {
     uint k = nbl_glsl_ext_FFT_calculateTwiddlePower(threadId, iteration, logTwoN);
-    return nbl_glsl_expImaginary(-1 * 2 * nbl_glsl_PI * k / N);
+    return nbl_glsl_expImaginary(-1.0f * 2.0f * nbl_glsl_PI * float(k) / (1 << logTwoN));
 }
 
-nbl_glsl_complex nbl_glsl_ext_FFT_twiddleInverse(in uint threadId, in uint iteration, in uint logTwoN, in uint N) 
+nbl_glsl_complex nbl_glsl_ext_FFT_twiddleInverse(in uint threadId, in uint iteration, in uint logTwoN) 
 {
-    return nbl_glsl_complex_conjugate(nbl_glsl_ext_FFT_twiddle(threadId, iteration, logTwoN, N));
+    return nbl_glsl_complex_conjugate(nbl_glsl_ext_FFT_twiddle(threadId, iteration, logTwoN));
+}
+
+uvec3 nbl_glsl_ext_FFT_getPaddedDimensions() {
+    nbl_glsl_ext_FFT_Parameters_t params = nbl_glsl_ext_FFT_getParameters();
+    return (params.padded_dimension.xyz);
+}
+uvec3 nbl_glsl_ext_FFT_getDimensions() {
+    nbl_glsl_ext_FFT_Parameters_t params = nbl_glsl_ext_FFT_getParameters();
+    return (params.dimension.xyz);
+}
+uint nbl_glsl_ext_FFT_getDirection() {
+    nbl_glsl_ext_FFT_Parameters_t params = nbl_glsl_ext_FFT_getParameters();
+    return (params.dimension.w >> 16) & 0x000000ff;
+}
+bool nbl_glsl_ext_FFT_getIsInverse() {
+    nbl_glsl_ext_FFT_Parameters_t params = nbl_glsl_ext_FFT_getParameters();
+    return bool((params.dimension.w >> 8) & 0x000000ff);
+}
+uint nbl_glsl_ext_FFT_getPaddingType() {
+    nbl_glsl_ext_FFT_Parameters_t params = nbl_glsl_ext_FFT_getParameters();
+    return (params.dimension.w) & 0x000000ff;
 }
```

Comment:
these I'd more to the parameters include and make them "methods" of the struct (just mangle the name)
`nbl_glsl_ext_FFT_Parameters_t_getDirection(nbl_glsl_ext_FFT_Parameters_t params)`

Its kinda bad when something has a custom bit encoding of various flags and properties and the "how they're packed" is not next to definition of the datatype/variable that holds them